olp-cpp-sdk-core/src/client/OlpClientSettingsFactory.cpp

```diff
@@ -42,7 +45,21 @@ OlpClientSettingsFactory::CreateDefaultNetworkRequestHandler(
 std::unique_ptr<cache::KeyValueCache>
 OlpClientSettingsFactory::CreateDefaultCache(cache::CacheSettings settings) {
   auto cache = std::make_unique<cache::DefaultCache>(std::move(settings));
-  cache->Open();
+  auto error = cache->Open();
+  if (error == cache::DefaultCache::StorageOpenResult::OpenDiskPathFailure) {
+    std::string disk_path_mutable = settings.disk_path_mutable
+                                        ? settings.disk_path_mutable.get()
+                                        : "(empty)";
+    std::string disk_path_protected = settings.disk_path_protected
+                                          ? settings.disk_path_protected.get()
+                                          : "(empty)";
```

Comment:
Optionally you can also use the exactly for this case exposed boost::optional::get_value_or(), means:

```cpp
auto disk_path_mutable = settings.disk_path_mutable.get_value_or("(empty)");
```